lib/matplotlib/backend_bases.py

```diff
@@ -2763,6 +2764,11 @@ def __init__(self, canvas):
         self.mode = ''  # a mode string for the status bar
         self.set_history_buttons()
 
+        @partial(canvas.mpl_connect, 'draw_event')
+        def define_home(event):
+            self.push_current()
+            canvas.mpl_disconnect(define_home)
```

Comment:
Ah, I now understand this and I like it :smile: 

I have not seen this pattern before, can you leave a short comment as to what this is doing?